src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agent.kenshoo.monitoring
+
+import java.lang
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics.{MetricRegistry, Timer}
+import org.mockito.{ArgumentMatcher, Mockito}
+import org.mockito.ArgumentMatchers.longThat
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito.{never, times, verify}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+
+class MonitoredHttpClientSpec extends UnitSpec with MockitoSugar with BeforeAndAfterEach {
+  val http = mock[HttpClient]
+  val httpAPIs = Map("/test/endpoint" -> "testEndpoint")
+  val testEndpointTimer = mock[Timer]
+  val newTimer = mock[Timer]
+
+  val kenshooRegistry = mock[MetricRegistry]
+
+  override def beforeEach(): Unit = {
+
+    super.beforeEach()
+    Mockito.reset(kenshooRegistry)
+    Mockito.reset(testEndpointTimer)
+  }
+
+
+  def greaterThanOrEqualTo(l: Long): ArgumentMatcher[lang.Long] = new ArgumentMatcher[lang.Long] {
+    override def matches(argument: lang.Long): Boolean = argument >= l
+  }
+
+  implicit val hc = HeaderCarrier()
+
+  "doGet" should {
+
+    "time the request if the url is whitelisted" in new MonitoredHttpClientTest(http, kenshooRegistry, httpAPIs){
```

Comment:
just create `new MonitoredHttpClientTest(http, kenshooRegistry, httpAPIs)` instance once and re use it as anyway you are resetting the mock